src/Arcus.BackgroundJobs.Tests.Unit/AzureActiveDirectory/ClientSecretExpirationInfoProviderTests.cs

```diff
@@ -0,0 +1,277 @@
+            Assert.Equal(credentials.KeyId, actualApplication.KeyId);
+            Assert.Equal(expirationDate, actualApplication.EndDateTime);
+            Assert.True(actualApplication.RemainingValidDays < 0);
+        }
+
+        [Fact]
+        public async Task GetApplicationsWithPotentialExpiredSecrets_WithExpiredSecrets_ReturnsAll()
+        {
+            // Arrange
+            DateTimeOffset expirationDate = BogusGenerator.Date.RecentOffset(days: 5);
+
+            IEnumerable<PasswordCredential> credentials = CreateStubbedCredentials(expirationDate);
+            Application[] applications = GetStubbedApplications(credentials);
+            GraphServiceClient client = CreateStubbedGraphClient(applications);
+            int expirationThresholdInDays = BogusGenerator.Random.Int(min: 1);
+
+            var provider = new ClientSecretExpirationInfoProvider(client, NullLogger.Instance);
+
+            // Act
+            IEnumerable<AzureApplication> actualApplications = await provider.GetApplicationsWithPotentialExpiredSecrets(expirationThresholdInDays);
+
+            // Assert
+            Assert.NotEmpty(actualApplications);
+            Assert.Equal(applications.SelectMany(app => app.PasswordCredentials).Count(), actualApplications.Count());
+        }
+
+        [Fact]
+        public async Task GetApplicationsWithPotentialExpiredSecrets_WithSoonExpiredSecrets_ReturnsAll()
+        {
+            // Arrange
+            DateTimeOffset expirationDate = BogusGenerator.Date.RecentOffset(days: 1);
+
+            IEnumerable<PasswordCredential> credentials = CreateStubbedCredentials(expirationDate);
+            Application[] applications = GetStubbedApplications(credentials);
+            GraphServiceClient client = CreateStubbedGraphClient(applications);
+            int expirationThresholdInDays = 1;
+
+            var provider = new ClientSecretExpirationInfoProvider(client, NullLogger.Instance);
+
+            // Act
+            IEnumerable<AzureApplication> actualApplications = await provider.GetApplicationsWithPotentialExpiredSecrets(expirationThresholdInDays);
+
+            // Assert
+            Assert.NotEmpty(actualApplications);
+            Assert.Equal(applications.SelectMany(app => app.PasswordCredentials).Count(), actualApplications.Count());
+        }
+
+        [Fact]
+        public async Task GetApplicationsWithPotentialExpiredSecrets_WithStillValidSecrets_ReturnsEmpty()
+        {
+            // Arrange
+            DateTimeOffset expirationDate = BogusGenerator.Date.SoonOffset(days: 5);
+            
+            IEnumerable<PasswordCredential> credentials = CreateStubbedCredentials(expirationDate);
+            IEnumerable<Application> applications = GetStubbedApplications(credentials);
+            GraphServiceClient client = CreateStubbedGraphClient(applications);
+            int expirationThresholdInDays = BogusGenerator.Random.Int(min: 1, max: 2);
```

Comment:
Woops, my bad. This test is failing bc it doesn't has the correct expiration date in days. You can change it to `2` for example.

```suggestion
            int expirationThresholdInDays = 2;
```